client/src/app/site/tags/components/tag-list.component.ts

```diff
@@ -0,0 +1,139 @@
+import { Component, OnInit } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import { Title } from '@angular/platform-browser';
+import { Tag } from '../../../shared/models/core/tag';
+import { ListViewBaseComponent } from '../../base/list-view-base';
+import { TagRepositoryService } from '../services/tag-repository.service';
+import { ViewTag } from '../models/view-tag';
+import { FormGroup } from '@angular/forms';
+
+/**
+ * Listview for the complete lsit of available Tags
+ * ### Usage:
+ * ```html
+ * <os-tag-list></os-tag-list>
+ * ```
+ */
+@Component({
+    selector: 'os-tag-list',
+    templateUrl: './tag-list.component.html',
+    styleUrls: ['./tag-list.component.css']
+})
+export class TagListComponent extends ListViewBaseComponent<ViewTag> implements OnInit {
+    /**
+     * Define the content of the ellipsis menu.
+     * Give it to the HeadBar to display them.
+     */
+    // public tagMenu = [
+    //     { // delete button, update button=
+    //         text: ' ',
+    //         icon: ' ',
+    //         action: ' '
+    //     }
+    // ];
+
+    public tagNewEdit: boolean;
+    public tagUpdateEdit: boolean;
+    public selectedTag: ViewTag;
+    public selectedTagCopy: ViewTag;
+
+    /**
+     *
+     * Constructor.
+     *
+     * @param repo the repository
+     * @param titleService
+     * @param translate
+     */
+    public constructor(private repo: TagRepositoryService, titleService: Title, translate: TranslateService) {
+        super(titleService, translate);
+        this.tagNewEdit = false;
+        this.tagUpdateEdit = false;
+    }
+
+    /**
+     * Init function.
+     * Sets the title, inits the table and calls the repo.
+     */
+    public ngOnInit(): void {
+        super.setTitle('Tags');
+        this.initTable();
+        this.repo.getViewModelListObservable().subscribe(newTags => {
+            this.dataSource.data = newTags;
+        });
+    }
+
+    public editOrCreate(): void {
+        if (this.selectedTag) {
+            this.onEditButton();
+        } else {
+            this.onPlusButton();
+        }
+    }
+    public onPlusButton(): void {
+        this.tagNewEdit = true;
+        this.tagUpdateEdit = false;
+    }
+
+    public onEditButton(): void {
+        this.tagNewEdit = false;
+        this.tagUpdateEdit = true;
+    }
+
+    /**
+     * Sends a new or updates tag to the dataStore
+     */
+    public saveTag(form: FormGroup): void {
+        if (form.value) {
+            if (this.repo.osNameUnique(form.value.name) === true) {
```

Comment:
The client does not care if the name is unique. The server will handle this for us.
also, this function would return boolean, checking `=== true` is unnecessary

Reply:
Wouldn't it be better to check for uniqueness before even sending it to the server, too?

For just a little more hassle on the client side, we could avoid a lot of traffic to the server by not even attempting to send bad data; instead of relying on the server to do all the filtering anyways.